src/dreyfus_index.erl

```diff
@@ -157,9 +157,27 @@ handle_call(info, _From, State) -> % obsolete
     Reply = info_int(State#state.index_pid),
     {reply, Reply, State}.
 
+handle_cast({ddoc_updated, DDocResult},
+            #state{
+                index=#index{sig=Sig, dbname=DbName}
+            }=State) ->
+    case check_if_index_is_deleted(DbName, DDocResult, Sig) of
+        true ->
+            erlang:send_after(index_shutdown_delay(), self(), index_deleted),
```

Comment:
think I said it before, but an arbitrary delay doesn't seem write. If a design document is updated such that there is no reference to this index, we should either let any requests to it finish and then delete the index or just delete it. The arbitrary delay is, I'm guessing, intended to avoid the errors caused by deleting an index while it's being queried? If so, we should at least document that it's a bit hackish.